cmd/dep/init.go

```diff
@@ -69,6 +69,12 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 		root = ctx.WorkingDir
 	} else {
 		root = args[0]
+		if !filepath.IsAbs(args[0]) {
+			root = filepath.Join(ctx.WorkingDir, args[0])
+		}
+		if err := os.MkdirAll(root, os.FileMode(0777)); err != nil {
```

Comment:
I had made it 755 in my previous PR, i got a comment to change it to 777. 
https://github.com/golang/dep/pull/549#discussion_r115897097